aspnetcore/migration/70-80.md

```diff
@@ -73,64 +73,541 @@ In the project file, update each [Microsoft.AspNetCore.*](https://www.nuget.org/
+   * [Update the target framework](#update-the-target-framework)
+   * [Update package references](#update-package-references)
 
-@code {
-    ...
+1. Move the contents of the `App` component (`App.razor`) to a new `Routes` component file (`Routes.razor`) added to the project's root folder. Leave the empty `App.razor` file in the app in the project's root folder.
 
-    private string BoundValue
-    {
-        get => ChildMessage ?? string.Empty;
-        set => ChildMessageChanged.InvokeAsync(value);
-    }
-}
-```
+1. Add an entry to the `_Imports.razor` file to make shorthand render modes available to the app:
 
-In .NET 7, you can use the new `@bind:get` and `@bind:set` modifiers to support two-way data binding and simplify the binding syntax:
+   ```razor
+   @using static Microsoft.AspNetCore.Components.Web.RenderMode
+   ```
 
-```razor
-<NestedGrandchild @bind-GrandchildMessage:get="ChildMessage" 
-    @bind-GrandchildMessage:set="ChildMessageChanged" />
-```
+1. Move the content in the `_Host` page (`Pages/_Host.cshtml`) to the empty `App.razor` file. Proceed to make the following changes to the `App` component.
 
-> [!IMPORTANT]
-> The `@bind:get` and `@bind:set` features are receiving further updates at this time. To take advantage of the latest updates, confirm that you've installed the [latest SDK](https://dotnet.microsoft.com/download/dotnet/7.0).
->
-> Using an event callback parameter (`[Parameter] public EventCallback<string> ValueChanged { get; set; }`) isn't supported. Instead, pass an <xref:System.Action>-returning or <xref:System.Threading.Tasks.Task>-returning method to `@bind:set`.
->
-> For more information, see the following resources:
->
-> * [Blazor `@bind:after` not working on .NET 7 RTM release (dotnet/aspnetcore #44957)](https://github.com/dotnet/aspnetcore/issues/44957)
-> * [`BindGetSetAfter701` sample app (javiercn/BindGetSetAfter701 GitHub repository)](https://github.com/javiercn/BindGetSetAfter701)
+   > [!NOTE]
+   > In the following example, the project's namespace is `Net8BlazorServer`. Adjust the namespace to match your project.
+
+   Remove the following lines from the top of the file:
+
+   ```diff
+   - @page "/"
+   - @using Microsoft.AspNetCore.Components.Web
+   - @namespace Net8BlazorServer.Pages
+   - @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+   ```
+   
+   Replace the preceding lines with the following:
+
+   ```razor
+   @inject IHostEnvironment Env
+   ```
+
+   Remove the tilde (`~`) from the `href` of the `<base>` tag:
+
+   ```diff
+   - <base href="~/" />
+   + <base href="/" />
+   ```
+
+   Remove the Component Tag Helper for the `HeadOutlet` component:
+
+   ```diff
+   - <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
+   ```
+
+   Replace the preceding line's tag helper with the `HeadOutlet` component:
+
+   ```razor
+   <HeadOutlet @rendermode="RenderMode.InteractiveServer" />
```

Comment:
Can we collapse this into a single step with the corresponding diff? Also, `RenderMode.InteractiveServer` can be simplified to just `InteractiveServer`.

````suggestion
   Remove the Component Tag Helper for the `HeadOutlet` component:

   ```diff
   - <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
   + <HeadOutlet @rendermode="InteractiveServer" />
   ```
````

Reply:
I'll do it, but I don't recommend it ... the dev loses the usefulness of the **Copy** button.

Reply:
... also ...

```
@rendermode="InteractiveServer"
```

... I thought that would be for RTM ... you're saying it works now under RC2?

Reply:
The simplified syntax is enabled by the `using static` for `RenderMode`, which you already tell the user to add in an earlier step. The Blazor Web App template will have this by default for the final .NET 8 release, but you can always add it yourself.

Reply:
Oh ... wait ... sorry ... it's VERY late here. I think you mean because we are going to have the ...

```razor
@using static Microsoft.AspNetCore.Components.Web.RenderMode
```

... here that we're good to go here. A template-based app wouldn't have it.

Ok ... gotcha.

I think the thing that I left out was adding it to the `.Client` project with a note to myself to remove that later. I'll take care of it.